File: Leetcode/2465. Number of Distinct Averages/solution.go
import (
	"sort"
)

func distinctAverages(nums []int) int {
	sort.Ints(nums)
	averageSeen := make(map[float64]bool)
	i, j := 0, len(nums)-1
	average := 0.0

	for i < j {
		average = float64(nums[i]+nums[j]) / 2.0
		averageSeen[average] = true
		i += 1
		j -= 1
	}

	return len(averageSeen)
}

File: Leetcode/2465. Number of Distinct Averages/solution.py
class Solution:
    def distinctAverages(self, nums: List[int]) -> int:
        i, j = 0, len(nums) - 1
        seen = set()
        nums.sort()

        while i < j:
            average = (nums[i] + nums[j]) / 2
            if average not in seen:
                seen.add(average)
            i += 1
            j -= 1
        
        return len(seen)
        